web/src/test/java/com/epam/rft/atsy/web/controllers/PasswordChangeControllerTest.java

```diff
@@ -0,0 +1,188 @@
+import com.epam.rft.atsy.service.domain.PasswordChangeDTO;
+import com.epam.rft.atsy.service.domain.PasswordHistoryDTO;
+import com.epam.rft.atsy.service.domain.UserDTO;
+import com.epam.rft.atsy.service.exception.PasswordValidationException;
+import com.epam.rft.atsy.service.passwordchange.validation.PasswordValidator;
+import com.epam.rft.atsy.service.security.UserDetailsAdapter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.web.method.annotation.AuthenticationPrincipalArgumentResolver;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.time.ZonedDateTime;
+import java.util.Date;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PasswordChangeControllerTest extends AbstractControllerTest {
+  private static final String REQUEST_URL = "/secure/password/manage";
+
+  private static final String VIEW_NAME = "password_change";
+
+  private static final Long USER_ID = 1L;
+
+  private static final String USER_NAME = "Test";
+
+  private static final String USER_PASSWORD = "pass";
+
+  private static final String PASSWORD_VALIDATION_ERROR_MESSAGE_KEY = "error.message.key";
+
+  private static final String VALIDATION_ERROR_KEY = "validationErrorKey";
+
+  private static final String VALIDATION_SUCCESS_KEY = "validationSuccessKey";
+
+  private static final String PASSWORDCHANGE_VALIDATION_SUCCESS =
+      "passwordchange.validation.success";
+
+  private static final String PASSWORD_NOT_SET = null;
+
+  @Mock
+  private PasswordChangeService passwordChangeService;
+
+  @Mock
+  private UserService userService;
+
+  @Mock
+  private PasswordValidator passwordValidator;
+
+  @InjectMocks
+  private PasswordChangeController passwordChangeController;
+
+  private PasswordChangeDTO passwordChangeDto;
+
+  private UserDetailsAdapter userDetailsAdapter;
+
+  @Override
+  protected Object[] controllersUnderTest() {
+    return new Object[]{passwordChangeController};
+  }
+
+  @Override
+  public void setUp() {
+    objectMapper = new ObjectMapper();
+
+    mockMvc =
+        MockMvcBuilders.standaloneSetup(controllersUnderTest())
+            .setViewResolvers(viewResolver())
+            .setControllerAdvice(controllerAdvice())
+            .setHandlerExceptionResolvers(uncheckedExceptionResolver())
+            .setValidator(localValidatorFactoryBean())
+            .setCustomArgumentResolvers(new AuthenticationPrincipalArgumentResolver())
```

Comment:
Move the settings into the AbstractControllerTest class.

Also check out this link as it can be an alternative solution:
http://docs.spring.io/spring-security/site/docs/4.0.x/reference/htmlsingle/#test-mockmvc-setup